src/datadog/tracer_config.h

```diff
@@ -97,6 +97,16 @@ struct TracerConfig {
   // `log_on_startup` is overridden by the `DD_TRACE_STARTUP_LOGS` environment
   // variable.
   bool log_on_startup = true;
+
+  // `trace_id_128_bit` indicates whether the tracer will generate and interpret
+  // the higher 64 bits of trace IDs. If true, the tracer will generate 128-bit
+  // trace IDs, extract them from incoming trace context, inject them into
+  // outgoing trace context, and send them to the collector. If false, then the
+  // tracer will generate 64-bit trace IDs, and extract only the lower 64 bits
+  // of trace IDs from incoming trace context.
+  // `trace_id_128_bit` is overridden by the `DD_TRACE_128_BIT_TRACEID_ENABLED`
+  // environment variable.
+  bool trace_id_128_bit = false;
```

Comment:
It would mean we always send the `_dd.p.tid` trace tag to the Agent and to forward services. The data cost is marginal, and I wasn't able to measure the latency cost. Still, the spec.